templates/nextflow-s3-template.yaml

```diff
@@ -0,0 +1,148 @@
+  OwnerEmail:
+    Description: 'Email address of the owner of this resource'
+    Type: String
+    AllowedPattern: '^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$'
+    ConstraintDescription: 'Must be an acceptable email address syntax(i.e. joe.smith@sagebase.org)'
+  BucketName:
+    Type: String
+    Description: (Optional) Name of the created bucket.
+  ReadWriteAccess:
+    Type: CommaDelimitedList
+    Default: "[]"
+    Description: (Optional) Grant read/write access to accounts, groups, and users.
+    ConstraintDescription: List of ARNs (i.e. ["arn:aws:iam::011223344556:user/jsmith", "arn:aws:iam::544332211006:user/rjones"]
+  ReadOnlyAccess:
+    Type: CommaDelimitedList
+    Default: "[]"
+    Description: (Optional) Grant read access to accounts, groups, and users.
+    ConstraintDescription: List of ARNs (i.e. ["arn:aws:iam::011223344556:user/jsmith", "arn:aws:iam::544332211006:user/rjones"]
+
+# Conditions
+Conditions:
+  HasBucketName: !Not [!Equals [!Ref BucketName, ""]]
+
+# Mappings
+Mappings:
+  Accounts:
+    "035458030717":
+      AdminRole: "arn:aws:iam::035458030717:role/aws-reserved/sso.amazonaws.com/AWSReservedSSO_Administrator_580e9f32ac55c4e7"
+    "728882028485":
+      AdminRole: "arn:aws:iam::728882028485:role/aws-reserved/sso.amazonaws.com/AWSReservedSSO_Administrator_3a9d61d0884be260"
+
+# Resources
+Resources:
+  Bucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !If [HasBucketName, !Ref BucketName, !Ref 'AWS::NoValue']
+      AccessControl: Private
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: AES256
+
+  # Policy
+  BucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref Bucket  
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+
+          - Sid: AllowDeleteAllowModify
+            # denies all except modifying, deleting and emptying bucket
+            Effect: Deny
+            Principal: '*'
+            NotAction:
+              - "s3:Delete*"
+              - "s3:GetBucket*"
+              - "s3:ListBucket*"
+              - "s3:PutBucket*"
+            Resource:
+              - !Sub arn:aws:s3:::${Bucket}
+              - !Sub arn:aws:s3:::${Bucket}/*
+
+          - Sid: AdminFullAccess
+            # gives admin full access to bucket
+            Effect: Allow
+            Principal:
+              AWS:
+                - !Sub "arn:aws:iam::${AWS::AccountId}:root"
+                - Fn::ImportValue: !Sub "${AWS::Region}-workflows-nextflow-ci-service-account-ServiceRoleArn"
+                - !FindInMap [Accounts, !Ref AWS::AccountId, AdminRole]
+            Action:
+              - "s3:*"
+            Resource:
+              - !Sub arn:aws:s3:::${Bucket}
+              - !Sub arn:aws:s3:::${Bucket}/*
+
+          - Sid: ReadWriteAccess
```

Comment:
This would be the statement where you would need to add [this condition](https://github.com/Sage-Bionetworks/aws-infra/blob/master/templates/s3-bucket-v2.j2#L233-L235).

Reply:
The bucket-owner-full-control parameter only applies to the "s3:*Object*" parameter, so it was applied in `Sid: ReadWriteAccess`. Everything else required for ReadWrite access was added separately in `Sid: ReadWriteBucketAccess`.